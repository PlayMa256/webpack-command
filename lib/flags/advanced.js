// ifArg("records-input-path", function(value) {
//   options.recordsInputPath = path.resolve(value);
// });
//
// ifArg("records-output-path", function(value) {
//   options.recordsOutputPath = path.resolve(value);
// });
//
// ifArg("records-path", function(value) {
//   options.recordsPath = path.resolve(value);
// });
//
// ifArg("target", function(value) {
//   options.target = value;
// });
//
// ifBooleanArg("hot", function() {
//   const HotModuleReplacementPlugin = require("webpack")
//     .HotModuleReplacementPlugin;
//   addPlugin(options, new HotModuleReplacementPlugin());
// });
//
// mapArgToBoolean("cache");
//
// mapArgToBoolean("bail");
//
// mapArgToBoolean("profile");
//
// ifArg("plugin", function(value) {
//   addPlugin(options, loadPlugin(value));
// });
//
// ifArg("prefetch", function(request) {
//   const PrefetchPlugin = require("webpack").PrefetchPlugin;
//   addPlugin(options, new PrefetchPlugin(request));
// });
//
// ifArg("provide", function(value) {
//   const idx = value.indexOf("=");
//   let name;
//   if (idx >= 0) {
//     name = value.substr(0, idx);
//     value = value.substr(idx + 1);
//   } else {
//     name = value;
//   }
//   const ProvidePlugin = require("webpack").ProvidePlugin;
//   addPlugin(options, new ProvidePlugin(name, value));
// });

// let defineObject;
// ifArgPair(
//   "define",
//   function(name, value) {
//     if (name === null) {
//       name = value;
//       value = true;
//     }
//     defineObject[name] = value;
//   },
//   function() {
//     defineObject = {};
//   },
//   function() {
//     const DefinePlugin = require("webpack").DefinePlugin;
//     addPlugin(options, new DefinePlugin(defineObject));
//   }
// );

// if (argv["watch-aggregate-timeout"]) {
//   options.watchOptions = options.watchOptions || {};
//   options.watchOptions.aggregateTimeout = +argv["watch-aggregate-timeout"];
// }
//
// if (typeof argv["watch-poll"] !== "undefined") {
//   options.watchOptions = options.watchOptions || {};
//   if (argv["watch-poll"] === "true" || argv["watch-poll"] === "")
//     options.watchOptions.poll = true;
//   else if (!isNaN(argv["watch-poll"]))
//     options.watchOptions.poll = +argv["watch-poll"];
// }
//
// if (argv["watch-stdin"]) {
//   options.watchOptions = options.watchOptions || {};
//   options.watchOptions.stdin = true;
//   options.watch = true;
// }

module.exports = {
  name: 'Advanced',
  flags: {
    bail: {
      type: 'boolean',
      desc: 'Abort the compilation on first error',
    },
    cache: {
      type: 'boolean',
      desc: 'Enable in memory caching',
    },
    define: {
      type: 'string',
      desc: 'Define any free var in the bundle',
    },
    hot: {
      type: 'boolean',
      desc: 'Enables Hot Module Replacement',
    },
    'labeled-modules': {
      type: 'boolean',
      desc: 'Enables labeled modules',
    },
    plugin: {
      type: 'string',
      desc: 'Load this plugin',
    },
    prefetch: {
      desc: 'Prefetch this request (Example: --prefetch ./file.js)',
      type: 'string',
    },
    profile: {
      desc: 'Profile the compilation and include information in stats',
      type: 'boolean',
    },
    provide: {
      desc:
        'Provide these modules as free vars in all modules (Example: --provide jQuery=jquery)',
      type: 'string',
    },
    'records-input-path': {
      desc: 'Path to the records file (reading)',
      type: 'string',
    },
    'records-output-path': {
      desc: 'Path to the records file (writing)',
      type: 'string',
    },
    'records-path': {
      desc: 'Path to the records file',
      type: 'string',
    },
    target: {
      desc: 'The targeted execution environment',
      type: 'string',
    },
    'watch-aggregate-timeout': {
      desc: 'Timeout for gathering changes while watching',
      // TODO: add type
    },
    'watch-poll': {
      desc: 'The polling interval for watching (also enable polling)',
      type: 'string',
    },
    'watch-stdin': {
      alias: 'stdin',
      desc: 'Exit the process when stdin is closed',
      type: 'boolean',
    },
  },
};
