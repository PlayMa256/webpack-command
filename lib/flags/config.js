// if (Array.isArray(options) && argv["config-name"]) {
//   const namedOptions = options.filter(function(opt) {
//     return opt.name === argv["config-name"];
//   });
//   if (namedOptions.length === 0) {
//     console.error(
//       "Configuration with name '" + argv["config-name"] + "' was not found."
//     );
//     process.exit(-1); // eslint-disable-line
//   } else if (namedOptions.length === 1) {
//     return processConfiguredOptions(namedOptions[0]);
//   }
//   options = namedOptions;
// }

// TODO: handle entry seperately, as it's not a flag
// ifArgPair(
//   "entry",
//   function(name, entry) {
//     if (
//       typeof options.entry[name] !== "undefined" &&
//       options.entry[name] !== null
//     ) {
//       options.entry[name] = [].concat(options.entry[name]).concat(entry);
//     } else {
//       options.entry[name] = entry;
//     }
//   },
//   function() {
//     ensureObject(options, "entry");
//   }
// );

/* eslint-disable no-param-reassign */

const merge = require('merge-options');

module.exports = {
  // eslint-disable-next-line no-unused-vars
  apply(argv, options) {
    let entry = {};

    // handle --entry file
    if (typeof argv.entry === 'string') {
      entry.cli = argv.entry;
    } else if (argv.entry) {
      // handle --entry.name file | --entry file --entry file2
      ({ entry } = argv);
    }

    if (Array.isArray(options.entry) || typeof options.entry === 'string') {
      // eslint-disable-next-line no-param-reassign
      options.entry = { main: options.entry };
    }

    entry = Object.assign(entry, options.entry);

    if (argv.configPath) {
      argv.require = argv.configPath;
    }

    options = merge(options, { mode: argv.mode, entry });
  },

  flags: {
    // TODO: handle config
    config: {
      desc: 'Path to the config file',
      type: 'string',
    },
    // TODO: handle config-name
    'config-name': {
      desc: 'Name of the config to use',
      type: 'string',
    },
    'config-register': {
      alias: 'r',
      desc:
        'Preload one or more modules before loading the webpack configuration',
      deprecated: '--require',
      type: ['string', 'array'],
    },
    mode: {
      // TODO: handle choices
      choices: ['development', 'production'],
      desc: 'Specifies the build mode to use; development or production',
      type: 'string',
    },
  },

  name: 'Configuration File',
};
