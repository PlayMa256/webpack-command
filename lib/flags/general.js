const path = require('path');

const chalk = require('chalk');
const merge = require('merge-options');
const webpack = require('webpack');

module.exports = {
  apply(argv, options) {
    // let entry = {};
    let plugins = [];
    const result = {};

    if (options.plugins) {
      plugins = plugins.concat(options.plugins);
    }

    if (argv.context) {
      result.context = path.resolve(argv.context);
    } else {
      result.context = process.cwd();
    }

    if (argv.debug) {
      const { LoaderOptionsPlugin } = webpack;
      const plugin = new LoaderOptionsPlugin({ debug: true });
      plugins.unshift(plugin);
    }

    if (argv.devtool) {
      result.devtool = argv.devtool;
    }

    if (argv.reporter) {
      result.reporter = argv.reporter;
    }

    // NOTE: runDev and runProd should be examined FIRST, as they maniuplate
    //       the argv object. maniuplating argv is bad and should make us feel
    //       bad.
    /* eslint-disable no-param-reassign */
    if (argv.runDev) {
      argv.debug = true;
      argv['output-pathinfo'] = true;
      if (!argv.devtool) {
        argv.devtool = 'eval-cheap-module-source-map';
      }
      if (!argv.mode) {
        argv.mode = 'development';
      }
    }

    if (argv.runProd) {
      argv['optimize-minimize'] = true;
      argv.define = []
        .concat(argv.define || [])
        .concat('process.env.NODE_ENV="production"');
      if (!argv.mode) {
        argv.mode = 'production';
      }
    }

    if (argv.watch) {
      result.watch = true;
    }

    return merge(options, result, plugins.length ? { plugins } : {});
  },

  flags: {
    context: {
      desc: 'The root directory for resolving entry point and stats',
      type: 'string',
    },
    debug: {
      desc: 'Switch loaders to debug mode',
      type: 'boolean',
    },
    devtool: {
      desc: chalk`Enable devtool for better debugging experience.
{dim e.g. --devtool eval-cheap-module-source-map}`,
      type: 'string',
    },
    entry: {
      desc: 'The entry point',
      type: ['array', 'object', 'string'],
    },
    help: {
      desc: 'Show usage information and the options listed here',
    },
    'log-level': {
      desc: chalk`Limit all process console messages to a specific level and above
{dim Levels: trace, debug, info, warn, error, silent}`,
      type: 'string',
    },
    'log-time': {
      desc:
        'Instruct the logger for webpack-serve and dependencies to display a timestamp',
    },
    reporter: {
      desc:
        'Specifies the reporter tpo use for generating console output for a build',
      type: 'string',
    },
    'run-dev': {
      alias: 'd',
      desc:
        'shortcut for --debug --devtool eval-cheap-module-source-map --output-pathinfo',
      type: 'boolean',
    },
    'run-prod': {
      alias: 'p',
      desc:
        'shortcut for --optimize-minimize --define process.env.NODE_ENV="production"',
      type: 'boolean',
    },
    version: {
      desc: 'Display the webpack-woof version',
    },
    watch: {
      alias: 'w',
      desc: 'Watch the filesystem for changes',
      type: 'boolean',
    },
  },

  name: 'General',
};
