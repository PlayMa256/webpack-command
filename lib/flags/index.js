'use strict';

const chalk = require('chalk');
const advanced = require('./advanced.js');
const config = require('./config.js');
const general = require('./general.js');
const modul = require('./module.js');
const optimization = require('./optimization.js');
const output = require('./output.js');
const resolver = require('./resolver.js');

const allFlags = [
  advanced,
  config,
  general,
  modul,
  optimization,
  output,
  resolver
];

function getLongest() {
  let result = 0;

  for (const group of allFlags) {
    for (const name of Object.keys(group.flags)) {
      const flag = group.flags[name];
      const alias = flag.alias ? `, -${flag.alias}` : '';
      result = Math.max(result, name.length + alias.length);
    }
  }

  return result;
}

function getGroupHelp(flags, maxLength) {
  const result = [];

  for (const name of Object.keys(flags)) {
    const flag = flags[name];
    const alias = flag.alias ? `, -${flag.alias}` : '';
    const diff = Math.abs((name.length + alias.length) - maxLength);
    const space = Array(diff + 2).fill(' ').join('');
    let { desc } = flag;
    const leftovers = desc.substring(80).split('');

    desc = desc.substring(0, 80);

    while (leftovers.length > 0) {
      const bits = [].concat('\n', Array(maxLength + 6).fill(' '), leftovers.splice(0, 80));
      desc += bits.join('');
    }

    result.push(`  --${name}${alias}${space}${desc}`);
  }

  return result;
}

module.exports = {
  help() {
    let result = [];
    const groups = allFlags.slice(0);
    const maxLength = getLongest();
    const generalHelp = getGroupHelp(general.flags, maxLength);

    result = result.concat(generalHelp, '');
    groups.splice(2, 1);

    for (const group of groups) {
      result.push(chalk`{underline ${group.name}}`);
      result = result.concat(getGroupHelp(group.flags, maxLength), '');
    }

    return result.join('\n');
  }
};
