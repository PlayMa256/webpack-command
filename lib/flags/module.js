const { bind } = require('./util');

module.exports = {
  apply(argv, options) {
    bind(argv.moduleBind, undefined, options); // eslint-disable-line no-undefined
    bind(argv.moduleBindPre, 'pre', options);
    bind(argv.moduleBindPost, 'post', options);
  },

  flags: {
    'module-bind': {
      desc: 'Bind an extension to a loader',
      type: 'string',
    },
    'module-bind-post': {
      // TODO: add description
      desc: '',
      type: 'string',
    },
    'module-bind-pre': {
      // TODO: add description
      desc: '',
      type: 'string',
    },
  },

  name: 'Modules',
};
