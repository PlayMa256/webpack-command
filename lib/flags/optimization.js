'use strict';

const merge = require('merge-options');
const { LoaderOptionsPlugin, optimize } = require('webpack');
const { validate } = require('./util');

const { MinChunkSizePlugin, LimitChunkCountPlugin } = optimize;

module.exports = {
  apply(argv, options) { // eslint-disable-line no-unused-vars
    options = merge({ plugins: [] }, options);

    const { flags } = module.exports;
    const keys = Object.keys(flags);

    for (const arg of keys) {
      const flag = flags[arg];
      const value = validate(flag, argv[arg]);

      if (value) {
        const plugin = flag.apply(value);
        options.plugins.unshift(plugin);
      }
    }
  },

  flags: {
    'optimize-max-chunks': {
      apply: value => new LimitChunkCountPlugin({ maxChunks: parseInt(value, 10) }),
      desc: 'Try to keep the chunk count below a limit',
      type: 'number'
    },
    'optimize-min-chunk-size': {
      apply: value => new MinChunkSizePlugin({ minChunkSize: parseInt(value, 10) }),
      desc: 'Try to keep the chunk size above a limit',
      type: 'number'
    },
    'optimize-minimize': {
      apply: () => new LoaderOptionsPlugin({ minimize: true }),
      desc: 'Minimize javascript and switches loaders to minimizing',
      type: 'boolean'
    }
  },

  name: 'Optimization'
};
