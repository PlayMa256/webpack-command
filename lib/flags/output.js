'use strict';

const path = require('path');
const camelcase = require('camelcase');
const merge = require('merge-options');
const { validate } = require('./util');

// if (argv.output) {
//   let output = argv.output;
//   if (!path.isAbsolute(argv.o)) {
//     output = path.resolve(process.cwd(), output);
//   }
//   argv["output-filename"] = path.basename(output);
//   argv["output-path"] = path.dirname(output);
// }

module.exports = {
  apply(argv, options) { // eslint-disable-line no-unused-vars
    const output = {};
    const { flags } = module.exports;
    const keys = Object.keys(flags).filter(key => key !== 'output');

    for (const arg of keys) {
      const flag = flags[arg];
      let value = argv[arg];

      value = validate(flag, value);

      if (arg === 'output-path') {
        value = path.resovle(value);
      }

      if (typeof value !== 'undefined') {
        output[camelcase(arg.replace('output-', ''))] = value;
      }
    }

    options = merge({ output }, options);
  },

  flags: {
    output: {
      alias: 'o',
      desc: 'The output path and file for compilation assets',
      type: 'string'
    },
    'output-chunk-filename': {
      desc: 'The output filename for additional chunks',
      type: 'string'
    },
    'output-filename': {
      desc: 'The output filename of the bundle',
      type: 'string'
    },
    'output-jsonp-function': {
      desc: 'The name of the jsonp function used for chunk loading',
      type: 'string'
    },
    'output-library': {
      desc: 'Expose the exports of the entry point as library',
      type: 'string'
    },
    'output-library-target': {
      desc: 'The type for exposing the exports of the entry point as library',
      type: 'string'
    },
    'output-path': {
      desc: 'The output path for compilation assets',
      type: 'string'
    },
    'output-pathinfo': {
      desc: 'Include a comment with the request for every dependency (require, import, etc.)',
      type: 'boolean'
    },
    'output-public-path': {
      desc: 'The public path for the assets',
      type: 'string'
    },
    'output-source-map-filename': {
      desc: 'The output filename for the SourceMap',
      type: 'string'
    }
  },

  name: 'Output'
};
