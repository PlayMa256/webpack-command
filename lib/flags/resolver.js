// function processResolveAlias(arg, key) {
//   ifArgPair(arg, function(name, value) {
//     if (!name) {
//       throw new Error("--" + arg + " <string>=<string>");
//     }
//     ensureObject(options, key);
//     ensureObject(options[key], "alias");
//     options[key].alias[name] = value;
//   });
// }
// processResolveAlias("resolve-alias", "resolve");
// processResolveAlias("resolve-loader-alias", "resolveLoader");
//
// ifArg("resolve-extensions", function(value) {
//   ensureObject(options, "resolve");
//   if (Array.isArray(value)) {
//     options.resolve.extensions = value;
//   } else {
//     options.resolve.extensions = value.split(/,\s*/);
//   }
// });

module.exports = {
  flags: {
    'resolve-alias': {
      desc:
        'Setup a module alias for resolving (Example: jquery-plugin=jquery.plugin)',
      type: 'string',
    },
    'resolve-extensions': {
      desc:
        'Setup extensions that should be used to resolve modules (Example: --resolve-extensions .es6,.js)',
      type: 'array',
    },
    'resolve-loader-alias': {
      desc: 'Setup a loader alias for resolving',
      type: 'string',
    },
  },

  name: 'Resolver',
};
