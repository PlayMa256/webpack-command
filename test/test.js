/* eslint-disable global-require */

const { register } = require('../lib/global');

register();

global.expect = require('expect');

require('./snapshot');

describe('Flags', () => {
  require('./tests/flags/bail');
  require('./tests/flags/config');
  require('./tests/flags/config-name');
  require('./tests/flags/context');
  require('./tests/flags/debug');
  require('./tests/flags/define');
});
