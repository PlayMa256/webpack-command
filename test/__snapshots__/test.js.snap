// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --bail > should apply #0 1`] = `
Object {
  "bail": true,
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/bail/entry.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --config > should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.85 KiB    main  [emitted]  main
Entrypoint main = main.js
[./src/index.js] 43 bytes {main} [built]"
`;

exports[`Flags > --context > should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>/test/fixtures/common",
  "entry": "./entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --context > should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.84 KiB    main  [emitted]  main
Entrypoint main = main.js
[./entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --debug > dev should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --debug > dev should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --debug > dev should build #1 1`] = `"bf58edd8"`;

exports[`Flags > --debug > prod should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "production",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --debug > prod should build #0 1`] = `
"Version: webpack 4.3.0
  Asset       Size  Chunks             Chunk Names
main.js  641 bytes       0  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {0} [built]"
`;

exports[`Flags > --debug > prod should build #1 1`] = `"969d6c1f"`;

exports[`Flags > --debug > should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    LoaderOptionsPlugin {
      "options": Object {
        "debug": true,
        "test": Object {
          "test": [Function],
        },
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --debug > should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --define > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/define/entry.js",
  "mode": "development",
  "plugins": Array [
    DefinePlugin {
      "definitions": Object {
        "DEFINE": "'valid'",
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
    DefinePlugin {
      "definitions": Object {
        "test": "invalid",
      },
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --define > should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  3.04 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/flags/define/entry.js] 91 bytes {main} [built]"
`;

exports[`Flags > --devtool > should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "devtool": "cheap-source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --devtool > should build #0 1`] = `
"Version: webpack 4.3.0
      Asset      Size  Chunks             Chunk Names
    main.js  2.92 KiB    main  [emitted]  main
main.js.map  2.64 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --devtool > should build #1 1`] = `"f349b18e"`;

exports[`Flags > --devtool > should build #2 1`] = `"195aef37"`;

exports[`Flags > --entry > multi should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "entry": Object {
    "main": Array [
      "./test/fixtures/common/entry-a.js",
      "./test/fixtures/common/entry-b.js",
    ],
  },
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --entry > multi should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  3.98 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]
[./test/fixtures/common/entry-b.js] 41 bytes {main} [built]
   [0] multi ./test/fixtures/common/entry-a.js ./test/fixtures/common/entry-b.js 40 bytes {main} [built]"
`;

exports[`Flags > --entry > multi should build #1 1`] = `"80e5fd27"`;

exports[`Flags > --entry > should parse flag array #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
  ],
}
`;

exports[`Flags > --entry > should parse flag object #0 1`] = `
Object {
  "a": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "b": "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
}
`;

exports[`Flags > --entry > should parse input #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-c.js",
  ],
}
`;

exports[`Flags > --entry > should parse input + flag array #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-c.js",
  ],
}
`;

exports[`Flags > --entry > should parse input + flag object #0 1`] = `
Object {
  "a": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "b": "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
  "main": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
}
`;

exports[`Flags > --entry > should parse input + single flag #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
  ],
}
`;

exports[`Flags > --entry > should parse single flag #0 1`] = `
Object {
  "main": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
}
`;

exports[`Flags > --entry > single should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "entry": Object {
    "main": "./test/fixtures/common/entry-a.js",
  },
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --entry > single should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --entry > single should build #1 1`] = `"bf58edd8"`;

exports[`Flags > --module-bind > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": undefined,
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind > should bind and build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.97 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 49 bytes {main} [built]"
`;

exports[`Flags > --module-bind > should bind and build #1 1`] = `"7d13f958"`;

exports[`Flags > --module-bind-all > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": undefined,
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader",
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "enforce": "pre",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-pre",
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "enforce": "post",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-post",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind-all > should bind and build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  3.01 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 86 bytes {main} [built]"
`;

exports[`Flags > --module-bind-all > should bind and build #1 1`] = `"64216e80"`;

exports[`Flags > --module-bind-post > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "post",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-post",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind-post > should bind and build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.98 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 60 bytes {main} [built]"
`;

exports[`Flags > --module-bind-post > should bind and build #1 1`] = `"c2baf8bb"`;

exports[`Flags > --module-bind-pre > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "pre",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-pre",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind-pre > should bind and build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.98 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 59 bytes {main} [built]"
`;

exports[`Flags > --module-bind-pre > should bind and build #1 1`] = `"f9b24195"`;

exports[`Flags > --optimize-* > max-chunks should apply #0 1`] = `
Object {
  "entry": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-c.js",
  ],
  "mode": "development",
  "plugins": Array [
    LimitChunkCountPlugin {
      "options": Object {
        "maxChunks": 1000,
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --optimize-* > max-chunks should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  4.67 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]
[./test/fixtures/common/entry-b.js] 41 bytes {main} [built]
[./test/fixtures/common/entry-c.js] 41 bytes {main} [built]
   [0] multi ./test/fixtures/common/entry-a.js ./test/fixtures/common/entry-b.js ./test/fixtures/common/entry-c.js 52 bytes {main} [built]"
`;

exports[`Flags > --optimize-* > max-chunks should build #1 1`] = `"adf33947"`;

exports[`Flags > --optimize-* > min-chunk-size should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    MinChunkSizePlugin {
      "options": Object {
        "minChunkSize": 10,
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --optimize-* > min-chunk-size should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --optimize-* > min-chunk-size should build #1 1`] = `"bf58edd8"`;

exports[`Flags > --optimize-* > minimize should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    LoaderOptionsPlugin {
      "options": Object {
        "minimize": true,
        "test": Object {
          "test": [Function],
        },
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --optimize-* > minimize should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --optimize-* > minimize should build #1 1`] = `"bf58edd8"`;
